### DISCUSSION TITLE
💥 Native preact/compat alias using only NPM

### DISCUSSION THREAD

**Discussion by unrevised6419** [preactjs/preact#3288] [Show and tell]
I'm playing around to move to Preact in a `create-react-app`.

In docs Preact says you need to alias `react` and `react-dom` to `preact/compat`.
`create-react-app` not allows aliases by default, you need to eject or use `craco` or `react-app-rewired

Then I remembered that NPM allows you to install a package under any name, using:

```shell
npm install new-name@npm:real-name
```

And I ralised that `npm:` is just the protocol, and we can use other protocols, like `file:`

**Install Preact**

```shell
npm install preact
```

**NPM Alias `react` and `react-dom` to `preact/compat`**
```shell
npm install react@file:node_modules/preact/compat
npm install react-dom@file:node_modules/preact/compat
```

**After install, remove `compat` types so TypeScript can consume `@types/react`**
```shell
npm set-script postinstall "rm node_modules/react/src/index.d.ts"
```

Boom 💥 -33.8KB from bundle
![image](https://user-images.githubusercontent.com/1881266/138378952-a1e83b3c-638b-43a3-9bce-374c8aeec9d1.png)

There is no need to configure your build system at all.
And it just works!

PR: https://github.com/roataway/roataway-web/pull/153
Build: https://app.netlify.com/sites/roataway/deploys/61721171aa8d4d00086a89d2
Preview: https://deploy-preview-153--roataway.netlify.app/

![](https://media.giphy.com/media/Uxma5LRUa5VAI/giphy.gif)

**Comment by piotr-cz** [comment]
Same effect should be simpler to accomplish with [@preact/compat](https://github.com/preactjs/compat-alias-package) package.
However I'm not sure how well it will play with *create-react-app*.
